File: src/RAG.Orchestrator.Api/Models/Feedback.cs
using System.ComponentModel.DataAnnotations;

namespace RAG.Orchestrator.Api.Models;

public class Feedback
{
    [Key]
    public Guid Id { get; set; }

    [Required]
    [MaxLength(450)]
    public string UserId { get; set; } = null!;

    [MaxLength(320)]
    public string? UserEmail { get; set; }

    [Required]
    [MaxLength(200)]
    public string Subject { get; set; } = null!;

    [Required]
    [MaxLength(4000)]
    public string Message { get; set; } = null!;

    [MaxLength(4000)]
    public string? Response { get; set; }

    [MaxLength(320)]
    public string? ResponseAuthorEmail { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime? RespondedAt { get; set; }
}

File: src/RAG.Web.UI/src/features/feedback/types/feedback.ts
export interface SubmitFeedbackRequest {
  subject: string
  message: string
  attachments: FeedbackAttachmentUpload[]
}

export interface RespondFeedbackRequest {
  response: string
}

export interface FeedbackItem {
  id: string
  userId: string
  userEmail?: string | null
  subject: string
  message: string
  response?: string | null
  responseAuthorEmail?: string | null
  createdAt: string
  updatedAt: string
  respondedAt?: string | null
  responseViewedAt?: string | null
  attachments: FeedbackAttachment[]
}

export interface FeedbackFilters {
  from?: string
  to?: string
  subject?: string
  userId?: string
  userEmail?: string
}

export interface FeedbackAttachment {
  id: string
  fileName: string
  contentType: string
  dataBase64: string
}

export interface FeedbackAttachmentUpload {
  fileName: string
  contentType: string
  dataBase64: string
}

